framework/database/__init__.py

```diff
@@ -31,7 +32,7 @@ def get_or_http_error(Model, pk_or_query, allow_deleted=False, display_name=None
     display_name = display_name or ''
     # FIXME: Not everything that uses this decorator needs to be markupsafe, but OsfWebRenderer error.mako does...
     safe_name = markupsafe.escape(display_name)
-    select_for_update = bool(request.method not in SAFE_METHODS)
+    select_for_update = bool(request.method not in SAFE_METHODS and not view_has_annotation(NO_AUTO_TRANSACTION_ATTR))
```

Comment:
It's still possible for this to be called outside of a transaction, i.e. if a view is managing it's own transactions.

```python
@no_auto_transaction
def myview():
    obj = get_or_http_error(..)
     with transaction.atomic():
            # ...
```

Does Django offer a way to check whether we're in the context of an active transaction?